Add new sections and articles (#244)
AdguardTeam/KnowledgeBaseDNS: docs/private-dns/connect-devices/other-options/automatic-connection.md

---
title: Automatic connection
sidebar_position: 5
---

## What it is and why you need it

> Not everyone feels at ease adding devices through the Dashboard. For instance, if you're a system administrator setting up multiple corporate devices simultaneously, you'd prefer to minimize manual tasks as much as possible.

Reviewer: ```suggestion
Not everyone feels at ease adding devices through the Dashboard. For instance, if you're a system administrator setting up multiple corporate devices simultaneously, you'll want to minimize manual tasks as much as possible.
```

Blah blah blah.

## How to configure Auto-connect devices

1. Open Dashboard.

1. Select the required *server*.

1. Go to *Devices*.

## Rules for generating automatically connected devices

### Device name

> When manually creating devices, please note that there are a number of restrictions related to name length, characters, spaces and hyphens:

Reviewer: ```suggestion
When manually creating devices, please note that there are a number of restrictions related to name length, characters, spaces, and hyphens:
```

**Name length**: maximum 50 characters. Characters beyond this limit are ignored.

**Permitted characters**: English letters, numbers and hyphens (`-`). Other characters are ignored.

**Spaces and hyphens**: Use a hyphen to indicate a space. For a hyphen, use a double hyphen `--`.

Let's look at one of the options with an example:

> - Let's create a device with the selected DNS-over-TLS protocol

Reviewer: ```suggestion
- Create a device with the selected DNS-over-TLS protocol
```

- Give it a space-sensitive name: AdGuard Test Device
- Device type: Android

> - Click the “Generate Link” button

Reviewer: ```suggestion
- Click the *Generate Link* button
```

sidebar_position: 5
---

## Why it is useful

Not everyone feels at ease adding devices through the Dashboard. For instance, if you’re a system administrator setting up multiple corporate devices simultaneously, you’ll want to minimize manual tasks as much as possible.

You can create a connection link and use it in the device settings. Your device will be detected and automatically connected to the server.

## How to configure automatic connection

1. Open the *Dashboard* and select the required server.
1. Go to *Devices*.
1. Enable the option to connect devices automatically.
    ![Connect devices automatically *mobile_border](https://cdn.adtidy.org/content/kb/dns/private/new_dns/connect/automatically_step4.png)

Now you can automatically connect your device to the server by creating a special address that includes the device name, device type, and current server ID. Let’s explore what these addresses look like and the rules for creating them.

### Examples of automatic connection addresses

> - `tls://adr-{Your_Server_ID}-AdGuard-Test-Device.d.adguard-dns.com` — This will automatically create an `Android` device with the `DNS-over-TLS` protocol named `AdGuard Test Device`.

Reviewer: тут точка есть, дальше нет
и подумать, начинать ли с большой буквы после тире и ставить ли тире в принципе

## What it is and why you need it

Not everyone feels at ease adding devices through the Dashboard. For instance, if you're a system administrator setting up multiple corporate devices simultaneously, you'll want to minimize manual tasks as much as possible.

You can create a connection link and use it in the device settings; it will be detected and automatically connect to the server.

In *Servers* → *Server settings* → *Devices*, we’ve added a template that generates a link for the specific device type and protocol. If you prefer to create links manually, just follow the instructions in the Dashboard.

## How to configure automatic connection

1. Open Dashboard.
1. Select the required server.
1. Go to *Devices*.

1. Enable the option to connect devices automatically.

    ![Connect devices automatically *border](https://cdn.adtidy.org/content/kb/dns/private/new_dns/connect/automatically_step4.png)

1. Next, go to the device settings in the *Connect devices automatically* section by clicking on the *Link generator and instructions* button

> 1. Select the desired protocol as well as the device name and type

Author: device type hehe
